src/components/RoomNameInput/index.js

```diff
@@ -1,68 +1,43 @@
-import React, {useState} from 'react';
-import _ from 'underscore';
+import React from 'react';
+import InputWrapper from '@components/Form/InputWrapper';
 import TextInput from '@components/TextInput';
 import useLocalize from '@hooks/useLocalize';
+import getOperatingSystem from '@libs/getOperatingSystem';
 import * as RoomNameInputUtils from '@libs/RoomNameInputUtils';
 import CONST from '@src/CONST';
 import * as roomNameInputPropTypes from './roomNameInputPropTypes';
 
-function RoomNameInput({isFocused, autoFocus, disabled, errorText, forwardedRef, value, onBlur, onChangeText, onInputChange, shouldDelayFocus}) {
+function RoomNameInput({isFocused, autoFocus, disabled, errorText, forwardedRef, onBlur, shouldDelayFocus, inputID, roomName}) {
     const {translate} = useLocalize();
 
-    const [selection, setSelection] = useState();
+    const keyboardType = getOperatingSystem() === CONST.OS.IOS ? CONST.KEYBOARD_TYPE.ASCII_CAPABLE : CONST.KEYBOARD_TYPE.VISIBLE_PASSWORD;
 
-    /**
-     * Calls the onChangeText callback with a modified room name
-     * @param {Event} event
-     */
-    const setModifiedRoomName = (event) => {
-        const roomName = event.nativeEvent.text;
-        const modifiedRoomName = RoomNameInputUtils.modifyRoomName(roomName);
-        onChangeText(modifiedRoomName);
-
-        // if custom component has onInputChange, use it to trigger changes (Form input)
-        if (_.isFunction(onInputChange)) {
-            onInputChange(modifiedRoomName);
-        }
-
-        // Prevent cursor jump behaviour:
-        // Check if newRoomNameWithHash is the same as modifiedRoomName
-        // If it is then the room name is valid (does not contain unallowed characters); no action required
-        // If not then the room name contains unvalid characters and we must adjust the cursor position manually
-        // Read more: https://github.com/Expensify/App/issues/12741
-        const oldRoomNameWithHash = value || '';
-        const newRoomNameWithHash = `${CONST.POLICY.ROOM_PREFIX}${roomName}`;
-        if (modifiedRoomName !== newRoomNameWithHash) {
-            const offset = modifiedRoomName.length - oldRoomNameWithHash.length;
-            const newSelection = {
-                start: selection.start + offset,
-                end: selection.end + offset,
-            };
-            setSelection(newSelection);
-        }
-    };
+    const valueParser = (innerRoomName) => RoomNameInputUtils.modifyRoomName(innerRoomName);
+    const displayParser = (innerRoomName) => RoomNameInputUtils.modifyRoomName(innerRoomName, true);
 
     return (
-        <TextInput
+        <InputWrapper
+            InputComponent={TextInput}
+            inputID={inputID}
```

Comment:
It would look better if we have moved the `InputWrapper` outside the component, but that would require dividing the logic of RoomNameInput between the component and screens where it's used. Since the logic of room input is quite complex and the current refactor has already caused some regressions in this PR I would only migrate the input to FormProvider. As soon as we will be sure the current changes won't introduce any regressions I would refactor the RoomNameInput according your suggestions in a separate PR.

Reply:
I'm trying to get rid of the `valueParser` and `displayParser` that we are passing to the Form. The `FromProvider` should not handle those. We should instead call the `onInputChange` in the `RoomNameInput` and supply the modified value.

This approach should result in the least code changes.


> but that would require dividing the logic of RoomNameInput between the component and screens

Can you please elaborate on this one?

Reply:
>I'm trying to get rid of the valueParser and displayParser that we are passing to the Form. The FromProvider should not handle those. We should instead call the onInputChange in the RoomNameInput and supply the modified value.

That would be great, but since we need to display different value than the one that is sent to Onyx, I couldn't find any better solution.

The second part I managed to move, I didn't spot that I forgot to move some props.

Reply:
@s77rt Ok, I reverted some changes, I think now it should look fine.

Reply:
Thanks. This is looking much better. I still see some changes to `src/components/RoomNameInput/` that does not look needed. And in `src/libs/RoomNameInputUtils.ts` too Can you please check again?

Reply:
Will do it tomorrow